embedding/csharp/FlutterApplication.cs

```diff
@@ -106,90 +105,89 @@ protected override void OnCreate()
+                foreach (string line in lines)
                 {
-                    EngineArgs.AddRange(lines);
+                    InternalLog.Info(LogTag, $"Enabled: {line}");
+                    EngineArgs.Add(line);
                 }
                 File.Delete(tempPath);
             }
             catch (Exception ex)
             {
-                InternalLog.Warn(LogTag, $"Error while processing a file:\n{ex}");
+                InternalLog.Warn(LogTag, $"Error while processing a file: {ex}");
             }
         }
 
         protected override void OnResume()
         {
             base.OnResume();
 
-            if (!Handle.IsInvalid)
-            {
-                FlutterNotifyAppIsResumed(Handle);
-            }
+            Debug.Assert(Handle);
+
+            FlutterNotifyAppIsResumed(Handle);
         }
 
         protected override void OnPause()
         {
             base.OnPause();
 
-            if (!Handle.IsInvalid)
-            {
-                FlutterNotifyAppIsPaused(Handle);
-            }
+            Debug.Assert(Handle);
+
+            FlutterNotifyAppIsPaused(Handle);
         }
 
         protected override void OnTerminate()
         {
             base.OnTerminate();
 
-            if (!Handle.IsInvalid)
-            {
-                FlutterDestroyWindow(Handle);
-            }
+            Debug.Assert(Handle);
+
+            FlutterDestroyWindow(Handle);
+        }
+
+        protected override void OnLowMemory(LowMemoryEventArgs e)
+        {
+            base.OnLowMemory(e);
+
+            Debug.Assert(Handle);
+
+            FlutterNotifyLowMemoryWarning(Handle);
         }
 
         protected override void OnLocaleChanged(LocaleChangedEventArgs e)
         {
             base.OnLocaleChanged(e);
 
-            if (!Handle.IsInvalid)
-            {
-                FlutterNotifyLocaleChange(Handle);
-            }
+            Debug.Assert(Handle);
+
+            FlutterNotifyLocaleChange(Handle);
         }
 
         protected override void OnRegionFormatChanged(RegionFormatChangedEventArgs e)
         {
             base.OnRegionFormatChanged(e);
 
-            if (!Handle.IsInvalid)
-            {
-                FlutterNotifyLocaleChange(Handle);
-            }
-        }
-
-        protected override void OnLowMemory(LowMemoryEventArgs e)
-        {
-            base.OnLowMemory(e);
+            Debug.Assert(Handle);
 
-            if (!Handle.IsInvalid)
-            {
-                FlutterNotifyLowMemoryWarning(Handle);
-            }
+            FlutterNotifyLocaleChange(Handle);
         }
 
-        public IntPtr GetPluginRegistrar(string pluginName)
+        /// <summary>
+        /// Returns the plugin registrar handle for the plugin with the given name.
+        /// The name must be unique across the application.
+        /// </summary>
+        public IntPtr GetRegistrarForPlugin(string pluginName)
         {
-            if (!Handle.IsInvalid)
+            if (Handle)
```

Comment:
Why didn't you use ```Debug.Assert(Handle)``` here?

Reply:
Conclusion: There's no exact reason.

Some client wrapper implementations for platforms like [glfw](https://github.com/flutter-tizen/engine/blob/31279eae37874150064da5983425c7769f0b0f40/shell/platform/glfw/client_wrapper/flutter_window_controller.cc#L81-L91) and windows explicitly check if the engine is currently running and emit error messages if not. Some platforms like [iOS](https://github.com/flutter-tizen/engine/blob/31279eae37874150064da5983425c7769f0b0f40/shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm#L237-L243) check if the instance is null but do not log errors, and some platforms like macOS just do not check anything. (And Tizen is the only platform that uses debug time asserts.) Sometimes things are different when it comes to lifecycle methods instead of plugin registry methods, but in general, the checks are more strict in case of plugin registry methods, maybe because each platform ensures `OnDestroy()` is called only when `OnCreate()` has succeeded.